src/Workspaces/Core/Portable/EmbeddedLanguages/RegularExpressions/LanguageServices/RegexPatternDetector.cs

```diff
@@ -0,0 +1,353 @@
+                {
+                    // Is a string argument to a method that looks like it could be a Regex method.  
+                    // Need to do deeper analysis
+                    var method = _semanticModel.GetSymbolInfo(invocationOrCreation, cancellationToken).GetAnySymbol();
+                    if (method != null &&
+                        method.DeclaredAccessibility == Accessibility.Public &&
+                        method.IsStatic &&
+                        _regexType.Equals(method.ContainingType))
+                    {
+                        return AnalyzeStringLiteral(
+                            stringLiteral, argumentNode, cancellationToken, out options);
+                    }
+                }
+            }
+            else if (syntaxFacts.IsObjectCreationExpression(invocationOrCreation))
+            {
+                var typeNode = syntaxFacts.GetObjectCreationType(invocationOrCreation);
+                var name = GetNameOfType(typeNode, syntaxFacts);
+                if (name != null)
+                {
+                    if (syntaxFacts.StringComparer.Compare(nameof(Regex), name) == 0)
+                    {
+                        var constructor = _semanticModel.GetSymbolInfo(invocationOrCreation, cancellationToken).GetAnySymbol();
+                        if (_regexType.Equals(constructor?.ContainingType))
+                        {
+                            // Argument to "new Regex".  Need to do deeper analysis
+                            return AnalyzeStringLiteral(
+                                stringLiteral, argumentNode, cancellationToken, out options);
+                        }
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        public RegexTree TryParseRegexPattern(SyntaxToken token, CancellationToken cancellationToken)
+        {
+            if (!this.IsRegexPattern(token, cancellationToken, out var options))
+            {
+                return null;
+            }
+
+            var chars = _language.VirtualCharService.TryConvertToVirtualChars(token);
+            if (chars.IsDefaultOrEmpty)
+            {
+                return null;
+            }
+
+            return RegexParser.TryParse(chars, options);
+        }
+
+        private bool AnalyzeStringLiteral(
+            SyntaxToken stringLiteral, SyntaxNode argumentNode, 
+            CancellationToken cancellationToken, out RegexOptions options)
+        {
+            options = default;
+
+            var parameter = _language.SemanticFacts.FindParameterForArgument(_semanticModel, argumentNode, cancellationToken);
+            if (parameter?.Name != _patternName)
+            {
+                return false;
+            }
+
+            options = GetRegexOptions(argumentNode, cancellationToken);
+            return true;
+        }
+
+        private RegexOptions GetRegexOptions(SyntaxNode argumentNode, CancellationToken cancellationToken)
```

Comment:
>GetRegexOptions [](start = 29, length = 15)

Similarly, it's probably good to cover this (I didn't see tests for `new Regex("pattern", ...some options...)`)

Reply:
i believe i have this.  but i'm fleshing this out more just in case :)